src/rules/cognitive-complexity.ts

```diff
@@ -357,26 +359,32 @@ const rule: TSESLint.RuleModule<string, (number | 'metric' | 'sonar-runtime')[]>
       }
     }
 
-    function isJsxShortCircuitNode(logicalExpression: TSESTree.LogicalExpression) {
-      const isShortCircuit =
-        logicalExpression.parent?.type === 'JSXExpressionContainer' &&
-        isJsxShortCircuitSubNode(logicalExpression, logicalExpression.left) &&
-        isJsxShortCircuitSubNode(logicalExpression, logicalExpression.right);
-      if (isShortCircuit) {
-        consideredLogicalExpressions.add(logicalExpression);
+    function getJsxShortCircuitNodes(logicalExpression: TSESTree.LogicalExpression) {
+      if (logicalExpression.parent?.type !== 'JSXExpressionContainer') {
+        return null;
+      } else {
+        return getJsxShortCircuitSubNodes(logicalExpression, logicalExpression);
       }
-      return isShortCircuit;
     }
 
-    function isJsxShortCircuitSubNode(root: TSESTree.LogicalExpression, node: TSESTree.Node) {
-      if (node.type === 'LogicalExpression') {
-        const isShortCircuit = node.operator === root.operator;
-        if (isShortCircuit) {
-          consideredLogicalExpressions.add(node);
-        }
-        return isShortCircuit;
+    function getJsxShortCircuitSubNodes(
+      root: TSESTree.LogicalExpression,
```

Comment:
maybe pass only the operator?

Reply:
Not sure about this as passing simply the node hides the implementation details that the check uses only the operator. Also, I don't think it improves anything in terms of performance, does it? But I guess it's simply a matter of preference.